docs/data/data-grid/column-menu/column-menu.md

```diff
@@ -0,0 +1,145 @@
+  />
+);
+```
+
+### Override default items and add new items:
+
+Using `componentsProps.columnMenu.columnMenuItems`, you can:
+
+1. Override Existing Items
+2. Add new Items
+
+```tsx
+const columnMenuItems = {
+  filter: <MyCustomFilter />, // override existing item
+  closeMenu: <MenuCloseComponent />, // add new item
+};
+
+// add new item in visible items and append it to the last of list
+const getVisibleColumnMenuItems = (defaultItems) => [...defaultItems, 'closeMenu'];
+
+<DataGrid
+  {...data}
+  componentProps={{ columnMenu: { getVisibleColumnMenuItems, columnMenuItems } }}
+/>;
+```
+
+If you're using TypeScript, for new items that you are adding, you'll need to specify new items you are registering, using [module augmentation](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation).
+
+```tsx
+declare module '@mui/x-data-grid' {
+  interface GridColumnMenuKeysOverrides {
+    closeMenu: true;
+  }
+}
+```
+
+### Default column menu items
+
+Default `keys` for **DataGrid** are `filter`, `sorting`, `hideColumn`, `divider` and `manageColumns`, wheras **DataGridPro** adds `pinning` and **DataGridPremium** adds `grouping` and `aggregation` on top of them.
+
+Here's a demo overriding some existing items, adding some new items and displaying different items for a column.
+
+{{"demo": "FilterColumnMenuGrid.js", "bg": "inline"}}
+
+## Column menu with Pro/Premium options [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan)[<span class="plan-premium"></span>](/x/introduction/licensing/#premium-plan)
+
+You can access commercial features like column pinning etc from the column menu when using `DataGridPro` or `DataGridPremium`.
+
+{{"demo": "ColumnMenuGridPro.js", "bg": "inline"}}
+
+## Simple column menu
+
+You can replace the default column menu with a simplified one using `GridColumnMenuSimple` component.
+
+{{"demo": "SimpleColumnMenuGrid.js", "bg": "inline"}}
+
+For _Pro_ and _Premium_ packages you have specific `slot` for each of the column menu item using which you can import simple variant or pass on your own component, or you can also use [column menu customizing](#customise-column-menu-items) to use your custom component based on item `key`.
+
+Here's a list of column menu items and their overridable slots for Commercial features.
+
+| **Item**                | **Package** | **Key**       | **Slot**                     | **Default**                     |
+| ----------------------- | ----------- | ------------- | ---------------------------- | ------------------------------- |
+| Pinning                 | Pro         | 'pinning'     | 'ColumnMenuPinningItem'      | GridColumnPinningMenuItems      |
+| Groupable (not grouped) | Premium     | 'grouping'    | 'ColumnMenuRowGroupableItem' | GridRowGroupableColumnMenuItems |
+| Grouping (grouped)      | Premium     | 'grouping'    | 'ColumnMenuRowGroupingItem'  | GridRowGroupingColumnMenuItems  |
```

Comment:
"Groupable" and "grouping" are a bit confusing, maybe use "group" and "ungroup" instead? What do you think?
```suggestion
| Group rows   | Premium     | 'grouping'    | 'ColumnMenuRowGroupableItem' | GridRowGroupableColumnMenuItems |
| Ungroup rows | Premium     | 'grouping'    | 'ColumnMenuRowGroupingItem'  | GridRowGroupingColumnMenuItems  |
```